src/System Application/App/File System/src/FileSystem/FileSystemImpl.Codeunit.al

```diff
@@ -0,0 +1,249 @@
+        CurrFileAccount: Record "File Account";
+        FileSystemConnector: Interface "File System Connector";
+        IsInitialized: Boolean;
+
+    procedure Initialize(Scenario: Enum "File Scenario")
+    var
+        FileAccount: Record "File Account";
+        FileScenarioMgt: Codeunit "File Scenario";
+        NoFileAccountFoundErr: Label 'No default file account defined.';
+    begin
+        if not FileScenarioMgt.GetFileAccount(Scenario, FileAccount) then
+            Error(NoFileAccountFoundErr);
+
+        Initialize(FileAccount);
+    end;
+
+    procedure Initialize(FileAccount: Record "File Account")
+    begin
+        CurrFileAccount := FileAccount;
+        FileSystemConnector := FileAccount.Connector;
+        IsInitialized := true;
+    end;
+
+    procedure ListFiles(Path: Text; FilePaginationData: Codeunit "File Pagination Data"; var FileAccountContent: Record "File Account Content" temporary)
+    begin
+        CheckPath(Path);
+        CheckInitialization();
+        FileSystemConnector.ListFiles(CurrFileAccount."Account Id", Path, FilePaginationData, FileAccountContent);
+    end;
+
+    procedure GetFile(Path: Text; Stream: InStream)
+    begin
+        CheckPath(Path);
+        CheckInitialization();
+        FileSystemConnector.GetFile(CurrFileAccount."Account Id", Path, Stream);
+    end;
+
+    procedure CreateFile(Path: Text; Stream: InStream)
+    begin
+        CheckPath(Path);
+        CheckInitialization();
+        FileSystemConnector.CreateFile(CurrFileAccount."Account Id", Path, Stream);
+    end;
+
+    procedure CopyFile(SourcePath: Text; TargetPath: Text)
+    begin
+        CheckPath(SourcePath);
+        CheckPath(TargetPath);
+        CheckInitialization();
+        FileSystemConnector.CopyFile(CurrFileAccount."Account Id", SourcePath, TargetPath);
+    end;
+
+    procedure MoveFile(SourcePath: Text; TargetPath: Text)
+    begin
+        CheckPath(SourcePath);
+        CheckPath(TargetPath);
+        CheckInitialization();
+        FileSystemConnector.MoveFile(CurrFileAccount."Account Id", SourcePath, TargetPath);
+    end;
+
+    procedure FileExists(Path: Text): Boolean
+    begin
+        CheckPath(Path);
+        CheckInitialization();
+        exit(FileSystemConnector.FileExists(CurrFileAccount."Account Id", Path));
+    end;
+
+    procedure DeleteFile(Path: Text)
+    begin
+        CheckPath(Path);
+        CheckInitialization();
+        FileSystemConnector.DeleteFile(CurrFileAccount."Account Id", Path);
+    end;
+
+    procedure ListDirectories(Path: Text; FilePaginationData: Codeunit "File Pagination Data"; var FileAccountContent: Record "File Account Content" temporary)
+    begin
+        CheckPath(Path);
+        CheckInitialization();
+        FileSystemConnector.ListDirectories(CurrFileAccount."Account Id", Path, FilePaginationData, FileAccountContent);
+    end;
+
+    procedure CreateDirectory(Path: Text)
+    begin
+        CheckPath(Path);
+        CheckInitialization();
```

Comment:
I'd swap these around.

Since the `CheckInitialization` is a small check and will throw an error if it fails, perform that first.